Classes/model/API/AlertAPIModels.swift

```diff
@@ -11,27 +11,30 @@ import Foundation
 extension API {
   
   public struct Alert: Codable {
-    
     public enum Severity: String, Codable {
       case info = "info"
       case warning = "warning"
       case alert = "alert"
     }
     
+    public struct Action: Codable {
+      let text: String
+      let type: String
```

Comment:
Should type be an enum? What are the possible values?

Reply:
at the moment Julieta only returned one value: "rerouteExcludingStops".

Reply:
Ok, please consider turning it into an enum, and also add tests that it's parsed correctly and that it handles unknown types for future proofness (it should still parse the alert content, but have a `nil` action).

Reply:
I've done this,

public struct Action: Codable {
      public enum Identifier: String, Codable {
        case reroute = "rerouteExcludingStops"
      }
      
      let text: String
      let type: Identifier?
      let excludedStopCodes: [String]?
}

But it seems you meant if `type` is nil, then `action` property on an `Alert` is nil too, even if it has a title? The doc hasn't been updated, so I'm not sure if we will have an action with just a title.

Having said that, how do you achieve this with `Codable`? It sounds like a custom implementation of `init(from decoder: Decoder)` is required - if `type` is nil due to the backend string fails to match enum value, then the init should return nil? But `init(from decoder: Decoder` cannot be failable.

Can you clarify/advise?

Reply:
Now I think about it, I feel the `Identifier` enum should really have an associated value. A `reroute` enum should have an array of stops being avoided. What do you think?

Let me investigate how this works with `Codable`.

Reply:
Yep, good idea on the associated value. You can do that with a custom `Codable`.

The initialiser is indeed not fallable, but it can throw. You should have a custom decodable on the Alert struct which is using the Action. Instead of the alert failing to parse and throwing, if the Action struct throws, it should then set action to nil (as it doesn't understand the type).